core/tests/test_utils.py

```diff
@@ -3030,12 +3033,35 @@ def get_response_without_checking_for_errors(
                 params, dict,
                 msg='Expected params to be a dict, received %s' % params)
 
+            if http_method != 'GET':
+                self.assertIn(
+                    http_method, ['POST', 'PUT', 'DELETE'])
+
         # This swap is required to ensure that the templates are fetched from
         # source directory instead of webpack_bundles since webpack_bundles is
         # only produced after webpack compilation which is not performed during
         # backend tests.
         with self.swap(base, 'load_template', mock_load_template):
-            response = self.testapp.get(url, params=params, expect_errors=True)
+
+            if http_method == 'GET':
+                response = self.testapp.get(
+                    url, params=params, expect_errors=True
+                )
+
+        if http_method == 'POST':
+            response = self.testapp.post(
+                url, params=params, expect_errors=True
+            )
+
+        elif http_method == 'PUT':
+            response = self.testapp.put(
+                url, params=params, expect_errors=True
+            )
+
+        elif http_method == 'DELETE':
```

Comment:
Perhaps change this to an "else" and assert in the next line that http_method is equal to DELETE.

Reply:
Tried that, but since this if block and the one for GET are separate, if the http_method is GET, it is checked again in this if block and this error is thrown:
```
Traceback (most recent call last):
  File "/home/runner/work/oppia/oppia/core/controllers/base_test.py", line 226, in test_that_no_get_results_in_500_error
    self.get_response_without_checking_for_errors(
  File "/home/runner/work/oppia/oppia/core/tests/test_utils.py", line 3060, in get_response_without_checking_for_errors
    self.assertIs(http_method, 'DELETE')
AssertionError: 'GET' is not 'DELETE'
``` 
So, changing it back to elif.

Reply:
OK, then do a "else raise Exception" after this?

Reply:
Done